Library/ViewModels/PledgeShippingLocationViewModel.swift

```diff
@@ -163,8 +163,13 @@ private func shippingValue(of project: Project, with shippingRuleCost: Double) -
 private func determineShippingRule(with project: Project, shippingRules: [ShippingRule]) -> ShippingRule? {
   guard
     let backing = project.personalization.backing,
-    let selectedRule = shippingRules.filter({ $0.location.id == backing.locationId }).first
+    let selectedRule = shippingRules.filter({ $0.location.id == backing.locationId }).first,
+    locationId(selectedRule.location.id, isValidIn: shippingRules)
```

Comment:
Let's say you picked Reward A and selected Canada as your shipping location. Now if you're changing to Reward B it will pre-select Canada, but only if Reward B _also_ ships to Canada. So the `locationId(_:isValidIn:)` check is this second bit of logic. If it fails it returns the default shipping rule.